### PR TITLE
fix: handle null and undefined in @fluentui/utilities shallowCompare

### LINKED ISSUE #25371: [Bug]: shallowCompare can attempt to get hasOwnProperty of null or undefined (leading to a JS exception)
### Library

React / v8 (@fluentui/react)

### System Info

```shell
windows, Edge
```

### Are you reporting Accessibility issue?

no

### Reproduction

https://codepen.io/jspurlin/pen/jOKOKbv

## Actual Behavior
passing `null` or `undefined` as the first parameter to `shallowCompare` causes an JS exception because it tries to get `hasOwnProperty` of `null` or `undefined`

## Expected Behavior
`shallowCompare` should be smart enough to handle `null` and/or `undefined` and not throw an exception

### Logs

_No response_

### Requested priority

Normal

### Products/sites affected

Office Online

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by smhigley**
Fixes #25371

Updates the `shallowCompare` in `@fluentui/utilities` to handle falsy values.

Previously, `shallowCompare(null, {a: 'b'})` would throw an error for trying to access `.hasOwnProperty` of `null`.

Now it does the following:
- If both compared variables are falsy it returns true regardless of if they're the same value, to preserve existing behavior. So `shallowCompare(null, undefined)` will return true (this was the case before the change).
- If one of the compared variables is falsy and the other is not, it will return false
- When both variables are truthy, the behavior is not changed

I also added some tests for `shallowCompare`, since it had none previously

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/b932f8ce3676d1fd82a6bdc79364007c9d3b2def'>b932f8ce3676d1fd82a6bdc79364007c9d3b2def</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>ComboBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/ComboBox-mount_base.html">1268</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/ComboBox-mount.html">1643</a></td>
            <td>1000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/ComboBox-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>GroupedList</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/GroupedList-mount_base.html">1919</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/GroupedList-mount.html">2178</a></td>
            <td>2</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/GroupedList-mount.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>BaseButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/BaseButton-mount_base.html">1221</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/BaseButton-mount.html">1206</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Breadcrumb</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/Breadcrumb-mount_base.html">2981</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/Breadcrumb-mount.html">2945</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Checkbox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/Checkbox-mount_base.html">2660</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/Checkbox-mount.html">2648</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>CheckboxBase</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/CheckboxBase-mount_base.html">2385</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/CheckboxBase-mount.html">2384</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ChoiceGroup</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/ChoiceGroup-mount_base.html">4472</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/ChoiceGroup-mount.html">4534</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ComboBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/ComboBox-mount_base.html">1268</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/ComboBox-mount.html">1643</a></td>
            <td>1000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/ComboBox-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>CommandBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/CommandBar-mount_base.html">9716</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/CommandBar-mount.html">9700</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ContextualMenu</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/ContextualMenu-mount_base.html">11765</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/ContextualMenu-mount.html">11317</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DefaultButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/DefaultButton-mount_base.html">1391</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/DefaultButton-mount.html">1382</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRow</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/DetailsRow-mount_base.html">3630</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/DetailsRow-mount.html">3651</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowFast</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/DetailsRowFast-mount_base.html">3684</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/DetailsRowFast-mount.html">3639</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowNoStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/DetailsRowNoStyles-mount_base.html">3470</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/DetailsRowNoStyles-mount.html">3509</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Dialog</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/Dialog-mount_base.html">3155</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/Dialog-mount.html">3136</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DocumentCardTitle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/DocumentCardTitle-mount_base.html">590</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/DocumentCardTitle-mount.html">594</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Dropdown</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/Dropdown-mount_base.html">3254</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/Dropdown-mount.html">3234</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusTrapZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/FocusTrapZone-mount_base.html">2023</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/FocusTrapZone-mount.html">2000</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/FocusZone-mount_base.html">1984</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/FocusZone-mount.html">1979</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedList</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/GroupedList-mount_base.html">1919</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/GroupedList-mount.html">2178</a></td>
            <td>2</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/GroupedList-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>GroupedList</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/GroupedList-virtual-rerender_base.html">1143</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/GroupedList-virtual-rerender.html">1142</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedList</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/GroupedList-virtual-rerender-with-unmount_base.html">1720</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/GroupedList-virtual-rerender-with-unmount.html">1698</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedListV2</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/GroupedListV2-mount_base.html">573</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/GroupedListV2-mount.html">595</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedListV2</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/GroupedListV2-virtual-rerender_base.html">556</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/GroupedListV2-virtual-rerender.html">568</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedListV2</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/GroupedListV2-virtual-rerender-with-unmount_base.html">586</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/GroupedListV2-virtual-rerender-with-unmount.html">589</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>IconButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/IconButton-mount_base.html">1909</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/IconButton-mount.html">1912</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Label</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/Label-mount_base.html">756</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/Label-mount.html">756</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Layer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/Layer-mount_base.html">4352</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/Layer-mount.html">4390</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Link</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/Link-mount_base.html">844</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/Link-mount.html">839</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MenuButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/MenuButton-mount_base.html">1704</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/MenuButton-mount.html">1701</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MessageBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/MessageBar-mount_base.html">2331</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/MessageBar-mount.html">2341</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Nav</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/Nav-mount_base.html">3296</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/Nav-mount.html">3301</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>OverflowSet</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/OverflowSet-mount_base.html">1373</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/OverflowSet-mount.html">1383</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Panel</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/Panel-mount_base.html">2573</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/Panel-mount.html">2588</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/Persona-mount_base.html">1322</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/Persona-mount.html">1303</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Pivot</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/Pivot-mount_base.html">1655</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/Pivot-mount.html">1638</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>PrimaryButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/PrimaryButton-mount_base.html">1527</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/PrimaryButton-mount.html">1551</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Rating</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/Rating-mount_base.html">7007</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/Rating-mount.html">7006</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SearchBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/SearchBox-mount_base.html">1584</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/SearchBox-mount.html">1534</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Shimmer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/Shimmer-mount_base.html">2879</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/Shimmer-mount.html">2927</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Slider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/Slider-mount_base.html">2138</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/Slider-mount.html">2119</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/SpinButton-mount_base.html">4675</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/SpinButton-mount.html">4714</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Spinner</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/Spinner-mount_base.html">812</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/Spinner-mount.html">807</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SplitButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/SplitButton-mount_base.html">3128</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/SplitButton-mount.html">3096</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Stack</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/Stack-mount_base.html">852</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/Stack-mount.html">853</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithIntrinsicChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/StackWithIntrinsicChildren-mount_base.html">2459</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/StackWithIntrinsicChildren-mount.html">2444</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithTextChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/StackWithTextChildren-mount_base.html">4910</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/StackWithTextChildren-mount.html">4889</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchColorPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/SwatchColorPicker-mount_base.html">10608</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/SwatchColorPicker-mount.html">10488</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TagPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/TagPicker-mount_base.html">2638</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/TagPicker-mount.html">2705</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TeachingBubble</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/TeachingBubble-mount_base.html">91578</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/TeachingBubble-mount.html">94399</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Text</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/Text-mount_base.html">810</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/Text-mount.html">801</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TextField</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/TextField-mount_base.html">1584</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/TextField-mount.html">1583</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/ThemeProvider-mount_base.html">1541</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/ThemeProvider-mount.html">1527</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/ThemeProvider-virtual-rerender_base.html">1094</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/ThemeProvider-virtual-rerender.html">1095</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/ThemeProvider-virtual-rerender-with-unmount_base.html">2151</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/ThemeProvider-virtual-rerender-with-unmount.html">2175</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Toggle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/Toggle-mount_base.html">1119</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/Toggle-mount.html">1100</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>buttonNative</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/buttonNative-mount_base.html">562</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25623/perf-test/buttonNative-mount.html">549</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9423"/>

File: packages/utilities/src/object.test.ts [hunk 1]
```diff
@@ -1,4 +1,55 @@
...
+  it('returns true for matching objects', () => {
+    const a = {
+      a: 1,
+      b: 'string',
+      c: {
+        d: 2,
+      },
+    };
+
+    const b = { ...a };
```
Comment (spmonahan) [inline packages/utilities/src/object.test.ts:13]: I think there is another variation of this test that is probably worth adding to the suite:

```ts
const a = {
  a: 1,
  b: 'string',
  c: {
    d: 2,
  },
};

const b = { ...a, c: {...a.c} };

expect(shallowCompare(a, b)).toBeFalsey();
```

Even though the objects have the same keys and values they are not equal to `shallowCompare()` because `a.c !== b.c`.

File: packages/utilities/src/object.ts [hunk 2]
```diff
@@ -5,6 +5,11 @@
  */
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 export function shallowCompare<TA extends any, TB extends any>(a: TA, b: TB): boolean {
+  if (!a || !b) {
+    // only return true if both a and b are falsy
+    return !a && !b;
```
[the remark this answers is not in the document]
Comment (smhigley) [reply packages/utilities/src/object.ts:10]: Yup, I did actually initially consider doing `return a === b` here for stuff like that, but opted to keep the existing behavior.

I'll add a test :)

**Review by spmonahan — APPROVED** [review]
Thanks for adding all the tests!
